week1/lc209_minimumSizeSubarraySum.js

```diff
@@ -26,4 +26,52 @@ var minSubArrayLen = function(s, nums) {
     }
     
     return (min < nums.length + 1)? min: 0;
-}
\ No newline at end of file
+}
+
+
+//follow up, nlogn solution
+/**
+ * @param {number} s
+ * @param {number[]} nums
+ * @return {number}
+ */
+var minSubArrayLen = function(s, nums) {
+    if(nums === null || nums.length === 0) return 0;
+    
+    let min = nums.length + 1;
+    let sums = new Array(nums.length);
+    
+    sums[-1] = 0;

```

Comment:
How about:
let sums = new Array(nums.length + 1)
sums[0] = 0;
for (let i = 1....)

Because negative index is not actually a `index`.
In your code, after giving sums[-1] = 0, what is the length of sums? Still `nums.length`.

Reply:
> How about:
> let sums = new Array(nums.length + 1)
> sums[0] = 0;
> for (let i = 1....)
> 
> Because negative index is not actually an `index`. Your `sums` becomes a compound datatype. It has an array with length of `nums.length`, and it has a property which the key is `-1` and value is `0`.
> In your code, after giving `sums[-1] = 0`, what is the length of `sums`? => Still `nums.length`. not `nums.length + 1`

I'm not sure if it's bad practice at the first beginning too, I wrote the code in this way just because it's more intuitive for me, I will try to revise that.

Reply:
It's ok here. You can keep this. 

I just want to point out we need to be careful when using the negative index.